src/Init/Data/Int/Basic.lean

```diff
@@ -325,6 +325,19 @@ protected def pow (m : Int) : Nat → Int
 instance : HPow Int Nat Int where
   hPow := Int.pow
 
+protected
+theorem pow_succ (b : Int) (n : Nat) : b ^ (n+1) = b^n * b := by rfl
+
+theorem powNatCast (b n : Nat) : (@Nat.cast Int _ b) ^ n = (b^n : Nat) := by
+  match n with
+  | 0 => rfl
+  | n + 1 =>
+    simp only [Int.pow_succ, powNatCast b n, Nat.pow_succ]
+    rfl
+
+theorem powOfNat (b n : Nat) : (@OfNat.ofNat Int b _) ^ n = (b^n : Nat) :=
+  powNatCast b n
```

Comment:
I think this should be called `ofNat_pow`?

Reply:
Hmm, actually not sure, I think a lot of our `ofNat_X` lemmas in the `Int` namespace need renaming, and probably should be `nat_cast_X` or `cast_X`.